blkid.c

```diff
@@ -255,8 +255,9 @@ static char *get_ubifs_dev_path(const char *dev, const char *vol,
 				const char *ubi_sys_path)
 {
 	glob_t files = { 0 };
-	char glob_exp[PATH_MAX] = { 0 };
-	snprintf(glob_exp, PATH_MAX, "%s/%s_*", ubi_sys_path, dev);
+	char glob_exp[PATH_MAX + NAME_MAX + 2 + 1] = { 0 };
```

Comment:
Not really, SNPRINTF_WTRUNC just truncate the output and log it, so we could have a problem with a truncated path. What the compilers see here is that `ubi_sys_path` has `PATH_MAX` so `glob_exp` needs to be bigger than `PATH_MAX`, then since we need to add a path, 2 chars ( _*) and the trailing 0 the new size makes sense

Reply:
We are talking about a path that is going to look like "/sys/devices/virtual/ubiX". It is not going to be bigger than PATH_MAX. Furthermore, it doesn't make sense to allocate paths that are bigger than PATH_MAX in general because of the limitation of the OS.